core/src/solution_submission.rs

```diff
@@ -358,6 +397,10 @@ mod tests {
             });
         // There should not be a second call to submit_solution because 90 to 100 is not a large
         // enough gas price increase.
+        contract
```

Comment:
This test doesn't really match with the comment above. I'm probably missing something, but don't we have to adjust the time somehow to trigger the noop transaction?